### PR TITLE
Refactor yearly spent function

### LINKED ISSUE #17: Refactor yearly spent function

### PR DESCRIPTION
**Pull request by Neil-B-Linton**
<h1>PULL REQUEST</h1>

<ul>
  <li>What does it change?</li>
Refactored the yearly spent, math wasn't accurate before..
  <li>What does it fix?</li>
Yearly spent function didn't account for the amount of travelers, the duration, or the 10% agent fee
  <li> Does it break any existing functionality or force me to update to a new version?</li>
no it fixes it 
</ul>

<h2>Type of Change</h2>

- [x] **Bug Fix**
- [x] **New Feature**
- [x] **New Change**

<h2>How has this been tested?</h2>

- [x] **JS**
- [x] **Dev Tools**
- [x] **Website**

<h2>Checklist</h2>

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new bugs
- [x] I have tagged project partners for review & it has been reviewed 

#17

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -90,18 +90,21 @@ const setTripsDestination = () => {
   let destinationObj = destinationRepo.findDestination(trip.destination);
   trip.destination = destinationObj;
   });
-}
+};
 
 const findYearlySpent = (travelersTrips) => {
   let yearlySpent = 0;
   travelersTrips.forEach(trip => {
```
Comment (alexmkio) [inline src/scripts.js]: this could be a reduce with an accumulator

File: src/scripts.js [hunk 2]
```diff
@@ -90,18 +90,21 @@ const setTripsDestination = () => {
...
     if(trip.date.includes('2022')) {
-      yearlySpent += parseInt(trip.destination.lodgingCost + trip.destination.flightCost)
+      let totalLodging = parseInt((trip.destination.lodgingCost * trip.travelers) * trip.duration)
+      let totalFlight =  parseInt(trip.destination.flightCost * trip.travelers)
+      let agentFee = (totalLodging + totalFlight) * .10
+      yearlySpent += totalLodging + totalFlight + agentFee
       displayYearlySpent(yearlySpent)
```
Comment (alexmkio) [inline src/scripts.js]: You have this function call inside AND outside of your iterator. Is that intentional?

File: src/scripts.js [hunk 3]
```diff
@@ -90,18 +90,21 @@ const setTripsDestination = () => {
   let destinationObj = destinationRepo.findDestination(trip.destination);
   trip.destination = destinationObj;
   });
-}
+};
 
 const findYearlySpent = (travelersTrips) => {
   let yearlySpent = 0;
   travelersTrips.forEach(trip => {
     if(trip.date.includes('2022')) {
```
Comment (alexmkio) [inline src/scripts.js:115]: Is there a more accurate and dynamic way you could find out if a trip happened "in the last year".

For example: will this function work as intended in 2023?

Look into [new Date()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date)

**Review by alexmkio — CHANGES_REQUESTED** [review]

File: src/scripts.js (hunk 3 above)
Comment (Neil-B-Linton) [reply src/scripts.js:115]: I have it hard coded at the moment, definitely not dynamic.. but I have to do more research for newDate()

File: src/scripts.js (hunk 1 above)
Comment (Neil-B-Linton) [reply src/scripts.js]: new push has a reduce, works way better than before

File: src/scripts.js (hunk 2 above)
Comment (Neil-B-Linton) [reply src/scripts.js]: this is fixed with the reduce.. now there isn't a need to call it twice

File: src/scripts.js (hunk 3 above)
Comment (alexmkio) [reply src/scripts.js:115]: Cool. Lets come back to this later!

**Comment by alexmkio** [comment]
Nice work! Lets come back later and make "last year" more dynamic.

**Review by alexmkio — APPROVED** [review]

**Pull request merged**